sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionClientTracer.java

```diff
@@ -0,0 +1,606 @@
+            writer.flush();
+            return new String(stream.toByteArray(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            logger.atWarning().log("'ChatRequestMessage' serialization error", e);
+        }
+        return null;
+    }
+
+    private String toJsonString(ChatChoice choice) {
+        try (ByteArrayOutputStream stream = new ByteArrayOutputStream();
+            JsonWriter writer = JsonProviders.createWriter(stream)) {
+            writer.writeStartObject();
+            writer.writeStartObject("message");
+            if (captureContent) {
+                writer.writeStringField("content", choice.getMessage().getContent());
+            }
+            if (choice.getMessage() != null) {
+                final List<ChatCompletionsToolCall> toolCalls = choice.getMessage().getToolCalls();
+                if (toolCalls != null && !toolCalls.isEmpty()) {
+                    writer.writeStartArray("tool_calls");
+                    for (ChatCompletionsToolCall toolCall : toolCalls) {
+                        if (captureContent) {
+                            toolCall.toJson(writer);
+                        } else {
+                            writer.writeStartObject();
+                            writer.writeStringField("id", toolCall.getId());
+                            writer.writeStringField("type", toolCall.getType());
+                            writer.writeEndObject();
+                        }
+                    }
+                    writer.writeEndArray();
+                }
+            }
+            writer.writeEndObject();
+            final CompletionsFinishReason finishReason = choice.getFinishReason();
+            if (finishReason != null) {
+                writer.writeStringField("finish_reason", finishReason.getValue());
+            }
+            writer.writeIntField("index", choice.getIndex());
+            writer.writeEndObject();
+            writer.flush();
+            return new String(stream.toByteArray(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            logger.atWarning().log("'ChatChoice' serialization error", e);
+        }
+        return null;
+    }
+
+    private static String getFinishReasons(List<ChatChoice> choices) {
+        final StringJoiner finishReasons = new StringJoiner(",", "[", "]");
+        for (ChatChoice choice : choices) {
+            final CompletionsFinishReason finishReason = choice.getFinishReason();
+            if (finishReason != null) {
+                finishReasons.add(finishReason.getValue());
+            }
+        }
+        return finishReasons.toString();
+    }
+
+    private static boolean isContentCapturingEnabled() {
+        final String envVal = System.getenv(OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT);
+        if ("true".equalsIgnoreCase(envVal)) {
+            return true;
+        }
+        final String propVal = System.getProperty(OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT);
```

Comment:
sys properties are usually lower-case, dot-separated. We should also use config instead of during direct calls similar to https://github.com/Azure/azure-sdk-for-java/blob/25dfa3de5300583a3986f29f9626efc640434230/sdk/core/azure-core/src/main/java/com/azure/core/util/TracingOptions.java#L17